docs/tutorial/fdp/index.rst

```diff
@@ -0,0 +1,179 @@
+
+There are 4 new log pages associated with **FDP**. These are FDP configuration,
+reclaim unit handle usage, FDP statistics and FDP events. All these log pages
+are endurance group scoped and hence you need to specify endurance group
+identifier in log specific identifier field.
+
+For all the above 4 mentioned log page, you can use **xNVMe** CLI:
+
+The fdp configuration log page requires dynamic memory allocation, as there
+can be multiple configuration, with each having multiple reclaim unit handles.
+You will have to specify the data size in bytes. The command can be run like:
+
+.. literalinclude:: 100_xnvme_log_fdp_config.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 100_xnvme_log_fdp_config.out
+   :language: bash
+
+For fdp statistics log page the command can be run like:
+
+.. literalinclude:: 200_xnvme_log_fdp_stats.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 200_xnvme_log_fdp_stats.out
+   :language: bash
+
+Similar to fdp configuration log page you will have to specify the number of
+recalim unit handle usage descriptors to fetch. The command can be run like:
+
+.. literalinclude:: 300_xnvme_log_ruhu.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 300_xnvme_log_ruhu.out
+   :language: bash
+
+The fdp events log page will have multiple events. You will have to specify
+number of events you want to fetch. You also need to specify whether you need
+host or controller events. This can be done by log specific parameter. The
+complete command can be run like:
+
+.. literalinclude:: 400_xnvme_log_fdp_events.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 400_xnvme_log_fdp_events.out
+   :language: bash
+
+Set and get-feature
+-------------------
+
+There are 2 new set and get-feature commands, introduced with FDP. These are
+Flexible data placement which controls operation of FDP capability in the
+specified endurance group, FDP events which controls if a controller generates
+FDP events associated with a specific reclaim unit handle.
+
+xNVMe does not support namespace management commands. Thus we cannot enable or
+disable FDP by sending set-feature command to the endurance group, as it
+requires deletion of all namespaces in that endurance group. However you can
+check the FDP capability by sending a get-feature command. The command can be
+run like:
+
+.. literalinclude:: 010_xnvme_feature_get.cmd
+   :language: bash
+
+The command should produce output similar to:
+
+.. literalinclude:: 010_xnvme_feature_get.out
+   :language: bash
+
+Command dword 12 controls whether you want to enable or disable FDP events.
+You will have to specify number of events to enable or disable and placement
+handle associated with it. These will be part of feat field. This will require
+you to specify the size in bytes of data buffer. To enable all the fdp events
```

Comment:
```suggestion
You will have to specify number of events to enable or disable and the Placement
Handle associated with it. These will be part of the Feat field. This will require
you to specify the size in bytes of the data buffer. To enable all the FDP Events
```